server/database.js

```diff
@@ -88,6 +88,13 @@ function getSubmissions(req, res, next) {
 }
 
 function approveAlgorithm(req, res, next) {
+	// Error if no user logged in or user not an admin
+	if (!"user" in req) {
+		return res.status(400).json("Must be logged in");
+	}
+	if ("user" in req && !req.user.is_admin) {
```

Comment:
if you get here, "user" has to be in req, right?

Reply:
Not if they are spoofing the post request

Reply:
I'm confused.  If user is ! in req, wouldn't you leave on line 93?

Reply:
Oh, I just misunderstood your comment when I replied above and forgot to update my comment